core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/RdsUpgradeOrchestratorService.java

```diff
@@ -52,6 +61,46 @@ public void restoreRdsData(Long stackId) throws CloudbreakOrchestratorException
         hostOrchestrator.runOrchestratorState(stateParams);
     }
 
+    public void upgradeEmbeddedDatabase(Long stackId) throws CloudbreakOrchestratorException {
+        OrchestratorStateParams stateParams = createStateParams(stackId, UPGRDE_EMBEDDED_DATABASE);
+        LOGGER.debug("Calling upgradeEmbeddedDatabase with state params '{}'", stateParams);
+        hostOrchestrator.runOrchestratorState(stateParams);
+    }
+
+    public void validateDbDirectorySpace(Long stackId) throws CloudbreakOrchestratorException {
+        OrchestratorStateParams stateParams = createStateParams(stackId, PREPARE_UPGRDE_EMBEDDED_DATABASE);
+        Map<String, String> dbVolumeSizeMap = hostOrchestrator.runCommandOnHosts(List.of(stateParams.getPrimaryGatewayConfig()), stateParams.getAllNodes(),
+                "df -k /dbfs | tail -1 | awk '{print $4}'");
+        Map<String, String> dataDirSizeMap = hostOrchestrator.runCommandOnHosts(List.of(stateParams.getPrimaryGatewayConfig()), stateParams.getAllNodes(),
+                "du -sk /dbfs/pgsql | awk '{print $1}'");
+        Integer dataDirSize = dataDirSizeMap.entrySet().stream()
+                .filter(e -> e.getKey().equals(stateParams.getPrimaryGatewayConfig().getHostname()))
```

Comment:
why it is useful to run on all the gateway nodes if we are only interested in the primary gateway's output? The command could be run only there instead.